scipy/stats/_qmc.py

```diff
@@ -1109,66 +1172,31 @@ class LatinHypercube(QMCEngine):
     def __init__(
         self, d: IntNumber, *, centered: bool = False,
         strength: int = 1,
-        optimization: Optional[Literal["random-cd"]] = None,
+        optimization: Optional[Literal["random-cd", "lloyd"]] = None,
         seed: SeedType = None
     ) -> None:
-        super().__init__(d=d, seed=seed)
+        super().__init__(d=d, seed=seed, optimization=optimization)
         self.centered = centered
 
         lhs_method_strength = {
-            1: self._random,
+            1: self._random_lhs,
             2: self._random_oa_lhs
         }
 
         try:
-            self.lhs_method = lhs_method_strength[strength]
+            self.lhs_method: Callable = lhs_method_strength[strength]
         except KeyError as exc:
             message = (f"{strength!r} is not a valid strength. It must be one"
                        f" of {set(lhs_method_strength)!r}")
             raise ValueError(message) from exc
 
-        optimization_method: Dict[Literal["random-cd"], Callable] = {
```

Comment:
This is all moved to QMCEngine or something, I imagine? What changes are needed?

Reply:
This part was moved to `_select_optimizer` which indeed is used in `QMCEngine`. This way I have a clear separation of responsibilities and I don't have a gigantic init. This way it's easier to read I think.